resolver/resolver.go

```diff
@@ -179,8 +161,71 @@ func (resolver *Resolver) CheckForCircularReferences() []*ResolvingError {
 	return resolver.resolvingErrors
 }
 
+func visitIndexWithoutDamagingIt(res *Resolver, idx *index.SpecIndex) {
+	mapped := idx.GetMappedReferencesSequenced()
+	mappedIndex := idx.GetMappedReferences()
+	res.indexesVisited++
+	for _, ref := range mapped {
+		seenReferences := make(map[string]bool)
+		var journey []*index.Reference
+		res.journeysTaken++
+		res.VisitReference(ref.Reference, seenReferences, journey, false)
+	}
+	schemas := idx.GetAllComponentSchemas()
+	for s, schemaRef := range schemas {
+		if mappedIndex[s] == nil {
+			seenReferences := make(map[string]bool)
+			var journey []*index.Reference
+			res.journeysTaken++
+			res.VisitReference(schemaRef, seenReferences, journey, false)
+		}
+	}
+	for _, c := range idx.GetChildren() {
+		visitIndexWithoutDamagingIt(res, c)
+	}
+}
+
+func visitIndex(res *Resolver, idx *index.SpecIndex) {
```

Comment:
why does this damage it? (versus the function above `visitIndexWithoutDamagingIt`)

Reply:
See: https://github.com/pb33f/libopenapi/blob/fix-73/resolver/resolver.go#L198

And: 

https://github.com/pb33f/libopenapi/blob/fix-73/resolver/resolver.go#L208

and

https://github.com/pb33f/libopenapi/blob/fix-73/resolver/resolver.go#L217

The resolver will destroy the root node by literally sticking together every reference found, inline, into a single giant mega tree. line numbers will be messed up, but the whole thing is walkable using code.

However the resolver isn't used to resolve by the datamodel, it's used by `vacuum`.

Reply:
ack; that makes sense now.